underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java

```diff
@@ -350,17 +350,24 @@ private String convertToFolderName(String key) {
    * @return true if the operation was successful, false otherwise
    */
   private boolean copy(String src, String dst) {
-    try {
-      src = stripPrefixIfPresent(src);
-      dst = stripPrefixIfPresent(dst);
-      LOG.info("Copying {} to {}", src, dst);
-      GSObject obj = new GSObject(dst);
-      mClient.copyObject(mBucketName, src, mBucketName, obj, false);
-      return true;
-    } catch (ServiceException e) {
-      LOG.error("Failed to rename file {} to {}", src, dst, e);
-      return false;
+    src = stripPrefixIfPresent(src);
+    dst = stripPrefixIfPresent(dst);
+    LOG.info("Copying {} to {}", src, dst);
+    GSObject obj = new GSObject(dst);
+    int retries = 3;
+    for (int i = 0; i < retries; ++i) {
```

Comment:
i think the Alluxio convention is i++. For primitives the difference syntax shouldn't have any performance impact.